src/explicit.rs

```diff
@@ -132,22 +143,27 @@ pub fn compute(
                 if stack.last().status != OverrideStatus::Isolate && stack.vec.len() >= 2 {
                     stack.vec.pop();
                 }
-                // The spec doesn't explicitly mention this step, but it is necessary.
-                // See the reference implementations for comparison.
+                // <https://www.unicode.org/reports/tr9/#Retaining_Explicit_Formatting_Characters>
                 levels[i] = stack.last().level;
+                processing_classes[i] = BN;
             }
 
             // Nothing
-            B | BN => {}
+            // BN case moved down to X6, see <https://www.unicode.org/reports/tr9/#Retaining_Explicit_Formatting_Characters>
+            B => {}
 
             // <http://www.unicode.org/reports/tr9/#X6>
             _ => {
                 let last = stack.last();
                 levels[i] = last.level;
-                match last.status {
-                    OverrideStatus::RTL => processing_classes[i] = R,
-                    OverrideStatus::LTR => processing_classes[i] = L,
-                    _ => {}
+                // This condition is not in the spec, but I am pretty sure that is a spec bug
+                // https://github.com/unicode-org/properties/issues/71
```

Comment:
That is a reference to a private repository (and it has been deduplicated there).

Link to your proposal instead?